pkg/gcv/validator.go

```diff
@@ -220,8 +222,60 @@ func (v *Validator) handleReview(ctx context.Context, idx int, asset *validator.
 	}
 }
 
+func ancestryPath(ancestors []string) string {
+	cnt := len(ancestors)
+	revAncestors := make([]string, len(ancestors))
+	for idx := 0; idx < cnt; idx++ {
+		revAncestors[cnt-idx-1] = ancestors[idx]
+	}
+	return strings.Join(revAncestors, "/")
+}
+
+const (
+	ancestryPathKey = "ancestry_path"
+	ancestorsKey    = "ancestors"
+)
+
+func (v *Validator) fixAncestry(input map[string]interface{}) error {
+	if _, found := input[ancestryPathKey]; found {
+		return nil
+	}
+
+	ancestorsIface, found := input[ancestorsKey]
+	if !found {
+		return errors.Errorf("no ancestors field: %s", input)
```

Comment:
I don't think we should raise an error here. Instead, I think we could just leave the function and not try to populate ancestry?

Reply:
I think we need to have something in the 'ancestry_path' field otherwise it won't get matched by anything (default match is "organizations/*").  Any objection to defaulting to "organizations/0" if we don't find the ancestors list?

Reply:
Right, or it might error out.

I mean strictly that there's no reason this function needs to error out if `ancestors` aren't provided. I'd prefer that we leave the logic as-is for cases where neither is provided.

Reply:
What about the mangled data case, eg, ancestors is the wrong type?  That seems like an error, but I guess we could log it for now.

Reply:
Agreed, that should be an error.

Reply:
done